test/signer.ts

```diff
@@ -429,12 +429,29 @@ describe('signer', () => {
           const qs1 = 'Z-Query=value';
           const qs2 = 'response-content-type=application%2Fjson';
 
-          assert(spy.firstCall.args[2].includes(`${qs1}&${qs2}`));
+          assert(spy.firstCall.returnValue.includes(`${qs1}&${qs2}`));
           assert(signedUrl.match(new RegExp(qs1)));
           assert(signedUrl.match(new RegExp(qs2)));
         });
       });
 
+      describe('v4 signed payload', () => {
+        it('shuold generate correct signed URL with a signed payload', async () => {
+          const spy = sandbox.spy(signer, 'getCanonicalRequest');
+
+          CONFIG.contentSha256 = 'sha256-hash';
+          CONFIG.version = 'v4';
+          const EXPECTED_HEADER = `x-goog-content-sha256:${CONFIG.contentSha256}`;
+
+          const signedUrl = await signer.getSignedUrl(CONFIG);
+
+          assert(spy.firstCall.returnValue.endsWith(CONFIG.contentSha256));
+          assert(spy.firstCall.args[3].includes(EXPECTED_HEADER));
+
+          assert(signedUrl.includes('X-Goog-SignedHeaders=host%3Bx-goog-content-sha256'));
```

Comment:
There's a conformance test for this specific case on https://github.com/googleapis/conformance-tests/blob/master/storage/v1/v4_signatures.json#L217

Reply:
Gotcha. I still would like a unit test to make sure proper things are passed around. Thanks.